src/components/reports/react/Charts/ApiResponseTime.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { useCallback, useEffect, useState } from "react";
+import { Body1, Subtitle2Stronger } from "@fluentui/react-components";
+import { IChartProps, LineChart } from "@fluentui/react-charting";
+import { Utils } from "../../../../utils";
+import { TReportsChartProps } from "../ReportsRuntime";
+
+const apiResponseTimeLineChartData: IChartProps = {
+    chartTitle: "API Response Time",
+    lineChartData: [
+        {
+            legend: "Minunum response time",
+            data: []
+        },
+        {
+            legend: "Average response time",
+            data: []
+        },
+        {
+            legend: "Maximum response time",
+            data: []
+        }
+    ]
+}
+
+export const ApiResponseTime = ({ reportsByTime, timeRange, dateFormattingFunc }: TReportsChartProps) => {
+    const [data, setData] = useState<IChartProps>(apiResponseTimeLineChartData);
+    const [forceRerender, setForceRerender] = useState<number>(1);
```

Comment:
How that forceRender is used?

Reply:
State change checks only "top-level values". In this case it doesn't trigger re-render because lineChartData's array inner array is changed. The top-level values and structure stays the same. The state doesn't check third-level values changes and thinks that the state is not changed. This is why the forceRerender is used.